numpyro/contrib/autoguide/__init__.py

```diff
@@ -366,6 +366,7 @@ def loss_fn(z):
         loc = params['{}_loc'.format(self.prefix)]
         precision = hessian(loss_fn)(loc)
         scale_tril = cholesky_inverse(precision)
+        scale_tril = np.where(np.isnan(scale_tril), 0., scale_tril)
```

Comment:
Do you think it is worth raising a warning here too?

Reply:
Sounds reasonable to me. I'll add it.